src/Serilog.Settings.Configuration/ConfigurationLoggerConfigurationExtensions.cs

```diff
@@ -38,19 +54,23 @@ public static class ConfigurationLoggerConfigurationExtensions
         /// </summary>
         /// <param name="settingConfiguration">Logger setting configuration.</param>
         /// <param name="configuration">A configuration object which contains a Serilog section.</param>
-        /// <param name="dependencyContext">The dependency context from which sink/enricher packages can be located. If not supplied, the platform
-        /// default will be used.</param>
+        /// <param name="dependencyContext">The dependency context from which sink/enricher packages can be located.</param>
+        /// <param name="onNullDependencyContext">If dependency context is not supplied, either the platform default or DLL scanning may be used.</param>
         /// <returns>An object allowing configuration to continue.</returns>
         public static LoggerConfiguration Configuration(
             this LoggerSettingsConfiguration settingConfiguration,
             IConfiguration configuration,
-            DependencyContext dependencyContext = null)
+            DependencyContext dependencyContext = null,
+            ConfigurationAssemblySource onNullDependencyContext = ConfigurationAssemblySource.UseLoadedAssemblies)
```

Comment:
To make this a non-breaking change, this'd need to be added in an overload, rather than a new, defaulted parameter (which is still a binary-breaking change). If you consider that the value is never used in conjunction with `dependencyContext`, the overloaded method signature could be:

```csharp
        public static LoggerConfiguration Configuration(
            this LoggerSettingsConfiguration settingConfiguration,
            IConfiguration configuration,
            ConfigurationAssemblySource configurationAssemblySource)
```

The non-default value would prevent overload resolution conflicts.

Reply:
Ah, much cleaner.